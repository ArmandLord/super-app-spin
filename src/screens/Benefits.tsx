import { Image, ScrollView, StyleSheet, View } from "react-native"
import Text from '../components/Text/Text'
import BaseCard from "../components/Card/components/BaseCard"
import StackedCardGrid from "../components/GridView/StackedCardGrid"
import { NativeStackScreenProps } from "@react-navigation/native-stack"
import { TStackBenefits } from "../types/navigation.types"
import { FEED, TRANSACTIONS } from "../constants/screens"

type Props = {}

export const Benefits = ({ navigation }: NativeStackScreenProps<TStackBenefits, typeof FEED>) => {
    return (
        <View style={styles.container}>
            <ScrollView>
                <StackedCardGrid containerStyle={{ flex: 0.5 }} titlesSize='small' numberOfColumns={2} idForTestContainer="grid" data={[
                    {
                        title: 'Consulta tu historial',
                        icon: <Image source={require('../assets/Images/star.png')} />,
                        onPress: () => navigation.navigate(TRANSACTIONS)
                    },
                    {
                        title: 'Cambia tus puntos',
                        icon: <Image source={require('../assets/Images/ticket.png')} />,
                        onPress: () => console.log("la 2")
                    },
                ]} />
                <BaseCard>
                    <Text testID="promo-1-text" variant="default-body-bold" numberOfLines={2}>
                        Acumula productos
                    </Text>
                    <Text variant="default-body-bold" numberOfLines={2}>
                        Muy pronto...
                    </Text>
                    <Image source={require('../assets/Images/star-card.png')} style={{ height: 50, width: 50 }} />
                </BaseCard>
                <BaseCard>
                    <Text testID="promo-2-text" variant="default-body-bold" numberOfLines={2}>
                        Gana m√°s puntos
                    </Text>
                    <Text variant="default-body-bold" numberOfLines={2}>
                        Muy pronto...
                    </Text>
                    <Image source={require('../assets/Images/star-plus.png')} style={{ height: 50, width: 50 }} />
                </BaseCard>
                <BaseCard>
                    <Text testID="promo-3-text" variant="default-body-bold" numberOfLines={2}>
                        Suma al comprar
                    </Text>
                    <Text variant="default-body-bold" numberOfLines={2}>
                        Muy pronto...
                    </Text>
                    <Image source={require('../assets/Images/star-ribbon.png')} style={{ height: 50, width: 50 }} />
                </BaseCard>
            </ScrollView>
        </View>
    )
}

const styles = StyleSheet.create({
    container: { flex: 1 }
})