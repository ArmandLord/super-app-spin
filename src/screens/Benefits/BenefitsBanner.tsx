import { Image, ScrollView } from 'react-native';
import styles from '../../styles/spinplus/Benefits/BenefitsBanner.styles';

const data = [
  require('../../assets/banner.png'),
  require('../../assets/banner.png'),
  require('../../assets/banner.png'),
  require('../../assets/banner.png'),
];

export const BenefitsBanner = () => (
  <ScrollView
    horizontal
    style={styles.container}
    contentContainerStyle={styles.banner}
    showsHorizontalScrollIndicator={false}>
    {data.map((image, index) => (
      <Image source={image} key={`img-${index}`} />
    ))}
  </ScrollView>
);
