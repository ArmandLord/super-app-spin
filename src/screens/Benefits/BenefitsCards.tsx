import { View, Image, useWindowDimensions, ImageProps } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import Card from '../../components/Card/Card';
import Text from '../../components/Text/Text';
import styles from '../../styles/spinplus/Benefits/BenefitsCards';

type BenefitCardT = {
  text: string;
  width: number;
  img: ImageProps;
  // REPLACE FOR SCREENS TYPE
  navigateTo: string;
};

// TODO: Add correct navigation on onPress action of the card when Movimientos and Cambio de puntos screen is ready
const BenefitCard = ({ img, text, navigateTo, width }: BenefitCardT) => {
  // ADD NAVIGATION HOOK WHEN MERGED NAVIGATION
  return (
    <View style={{ width }}>
      <Card style={styles.cardContainer} testID="benefit-card">
        <TouchableOpacity>
          <Image style={styles.img} width={124} height={124} source={img} />
          <View style={styles.cardTextContainer}>
            <Text variant="small-body-bold" style={styles.cardText}>
              {text}
            </Text>
          </View>
        </TouchableOpacity>
      </Card>
    </View>
  );
};

export const BenefitsCards = () => {
  const { width } = useWindowDimensions();
  const twoColumnsCardWidth = width / 2 - 24;

  return (
    <View style={styles.container} testID="benefit-cards">
      <BenefitCard
        navigateTo=""
        width={twoColumnsCardWidth}
        text={`Consulta \n tu historial`}
        img={require('../../assets/benefits-history.png')}
      />
      <BenefitCard
        navigateTo=""
        width={twoColumnsCardWidth}
        text="Cambia tus puntos"
        img={require('../../assets/change-points.png')}
      />
    </View>
  );
};
