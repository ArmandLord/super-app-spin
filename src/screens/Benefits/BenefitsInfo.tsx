import { View, Image } from 'react-native';
import Text from '../../components/Text/Text';
import styles from '../../styles/spinplus/Benefits/BenefitsInfo.styles';

// TODO: Replace hardcoded text with retrieved from endpoint
export const BenefitsInfo = () => {
  return (
    <View style={styles.container} testID="benefits-info">
      <View>
        <Text style={styles.pointsText} variant="headline-large">
          10, 657 puntos
        </Text>
        <View style={styles.pointsContainer}>
          <Image
            width={16}
            height={16}
            source={require('../../assets/starburst.png')}
          />
          <Text style={styles.pointsValueText} variant="label-extra-small-bold">
            Valen $156.00
          </Text>
        </View>
      </View>
      <Image
        width={120}
        height={120}
        source={require('../../assets/spinPremiaRibbon.png')}
      />
    </View>
  );
};
