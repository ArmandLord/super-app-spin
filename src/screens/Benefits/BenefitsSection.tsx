import { Image, ImageProps, View } from 'react-native';
import Text from '../../components/Text/Text';
import styles from '../../styles/spinplus/Benefits/BenefitsSection.styles';

type PropsT = {
  text: string;
  title: string;
  img: ImageProps;
};
export const BenefitsSection = ({ title, text, img }: PropsT) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title} variant="headline-small">
        {title}
      </Text>
      <Text style={styles.paragraph}>{text}</Text>
      <Image style={styles.img} source={img} />
    </View>
  );
};
