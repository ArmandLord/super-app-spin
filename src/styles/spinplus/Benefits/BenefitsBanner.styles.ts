import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  container: {
    flexGrow: 0,
  },
  banner: {
    gap: 16,
    paddingVertical: 24,
  },
});

export default styles;
