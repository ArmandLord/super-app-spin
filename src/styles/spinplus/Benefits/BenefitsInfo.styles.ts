import { StyleSheet } from 'react-native';
import { getColors } from '../..';
import { ThemeVariant } from '../../../theme/types';

const colors = getColors(true, ThemeVariant.SpinPlus);

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  pointsContainer: {
    flexDirection: 'row',
    paddingVertical: 4,
    borderRadius: 12,
    marginTop: 8,
    alignSelf: 'flex-start',
    backgroundColor: colors.surface_points,
    paddingHorizontal: 12,
  },
  pointsText: {
    fontWeight: '700',
  },
  pointsValueText: {
    marginLeft: 8,
    color: colors.content_points,
    fontWeight: '600',
  },
});

export default styles;
