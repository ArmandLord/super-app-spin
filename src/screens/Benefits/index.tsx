import { ScrollView } from 'react-native';
import { BenefitsInfo } from './BenefitsInfo';
import { BenefitsCards } from './BenefitsCards';
import { BenefitsSection } from './BenefitsSection';
import styles from '../../styles/spinplus/Benefits/Benefits.styles';

export const Benefits = () => {
  return (
    <ScrollView
      style={styles.container}
      contentContainerStyle={styles.contentContainer}>
      <BenefitsInfo />
      <BenefitsCards />
      <BenefitsSection
        title="Acumula produtos"
        img={require('../../assets/acumulaProductos.png')}
        text="Muy pronto podrás sumar tus compras y ganar productos de regalo"
      />
      <BenefitsSection
        title="Gana más puntos"
        img={require('../../assets/ganaPuntos.png')}
        text="Muy pronto podrás ganar más puntos en el total de tu compra"
      />
      <BenefitsSection
        title="Suma al comprar"
        img={require('../../assets/sumaAlComprar.png')}
        text="Muy pronto podrás acumular compras y llevarte productos de regalo"
      />
    </ScrollView>
  );
};
