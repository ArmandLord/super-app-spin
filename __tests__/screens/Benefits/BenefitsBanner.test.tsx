import React from 'react';
import { render } from '@testing-library/react-native';
import { BenefitsBanner } from '../../../src/screens/Benefits/BenefitsBanner';

describe('Benefits Banner', () => {
  it('renders correctly', () => {
    const { toJSON } = render(<BenefitsBanner />);
    expect(toJSON()).toMatchSnapshot();
  });
});
