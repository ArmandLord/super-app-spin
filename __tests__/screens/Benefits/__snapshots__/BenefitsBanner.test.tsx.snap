// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Benefits Banner renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "gap": 16,
      "paddingVertical": 24,
    }
  }
  data={
    [
      {
        "testUri": "../../../src/assets/banner.png",
      },
      {
        "testUri": "../../../src/assets/banner.png",
      },
      {
        "testUri": "../../../src/assets/banner.png",
      },
      {
        "testUri": "../../../src/assets/banner.png",
      },
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={true}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsHorizontalScrollIndicator={false}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/banner.png",
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/banner.png",
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/banner.png",
          }
        }
      />
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={
        [
          {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <Image
        source={
          {
            "testUri": "../../../src/assets/banner.png",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
